### PR TITLE
Add support for kubeconfig flag

### PR DESCRIPTION
**Pull request by rossf7**
Adds kubeconfig support for use with `kind`. See https://github.com/giantswarm/operatorkit/pull/289 for the related operatorkit change.

- Creating a cluster.

```bash
go get -u sigs.k8s.io/kind && kind create cluster 
```

- Running locally.

```bash
kubeconfig=`cat $(kind get kubeconfig-path --name="kind")` && \
go run main.go daemon --service.kubernetes.kubeconfig="$kubeconfig"
```

https://github.com/kubernetes-sigs/kind

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -65,7 +65,7 @@ required = [
   name = "github.com/giantswarm/micrologger"
 
 [[constraint]]
-  branch = "master"
+  branch = "kubeconfig"
```
Comment (rossf7) [inline Gopkg.toml]: TODO

File: flag/service/kubernetes/kubernetes.go [hunk 2]
```diff
@@ -8,8 +8,9 @@ import (
...
 type Kubernetes struct {
-	Address   string
-	InCluster string
-	TLS       tls.TLS
-	Watch     watch.Watch
+	Address    string
+	InCluster  string
+	KubeConfig string
+	TLS        tls.TLS
+	Watch      watch.Watch
```
Comment (xh3b4sd) [inline flag/service/kubernetes/kubernetes.go]: What do you think about having the common flags in `operatorkit` and using the structures in the operators' flag package by importing? We could then be consistent via vendor updates.

Comment (rossf7) [reply flag/service/kubernetes/kubernetes.go]: Yes it makes sense to move the flags to operatorkit. I'd like to add the kubeconfig support to all operators so I can use the vendored flags then.

Comment (rossf7) [reply flag/service/kubernetes/kubernetes.go]: Done. PTAL.

**Pull request merged**